target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-magic-2-wifi-next.dts

```diff
@@ -0,0 +1,269 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qcom-ipq4019.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "devolo Magic 2 WiFi next";
+	compatible = "devolo,magic-2-wifi-next";
+
+	memory {
+		device_type = "memory";
+		reg = <0x80000000 0x10000000>;
+	};
+
+	soc {
+		rng@22000 {
+			status = "okay";
+		};
+
+		mdio@90000 {
+			status = "okay";
+			pinctrl-0 = <&mdio_pins>;
+			pinctrl-names = "default";
+			reset-gpios = <&tlmm 59 GPIO_ACTIVE_LOW>;
+			reset-delay-us = <2000>;
+
+			/delete-node/ ethernet-phy@0;
+			/delete-node/ ethernet-phy@1;
+		};
+
+		ess-psgmii@98000 {
+			status = "okay";
+		};
+
+		crypto@8e3a000 {
+			status = "okay";
+		};
+
+		watchdog@b017000 {
+			status = "okay";
+		};
+
+		ess-switch@c000000 {
+			status = "okay";
+			switch_lan_bmp = <0x3e>;
+			switch_wan_bmp = <0x0>;
+		};
+
+		edma@c080000 {
+			status = "okay";
+			qcom,num_gmac = <3>;
+
+			gmac0 {
+				qcom,phy_mdio_addr = <3>;
+				qcom,poll_required = <1>;
+				/delete-property/ qcom,forced_speed;
+				/delete-property/ qcom,forced_duplex;
+				vlan_tag = <1 0x10>;
+			};
+
+			gmac1 {
+				qcom,phy_mdio_addr = <2>;
+				qcom,poll_required = <1>;
+				/delete-property/ qcom,forced_speed;
+				/delete-property/ qcom,forced_duplex;
+				vlan_tag = <1 0x08>;
+			};
+
+			gmac2 {
+				local-mac-address = [00 00 00 00 00 00];
+				qcom,phy_mdio_addr = <4>;
+				qcom,poll_required = <1>;
+				/delete-property/ qcom,forced_speed;
+				/delete-property/ qcom,forced_duplex;
+				vlan_tag = <1 0x20>;
+			};
+		};
+
+		gpio_export {
+			compatible = "gpio-export";
+			#size-cells = <0>;
+
+			plc {
+				gpio-export,name = "plc-enable";
+				gpio-export,output = <1>;
+				gpios = <&tlmm 63 GPIO_ACTIVE_HIGH>;
+			};
+		};
+
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		wlan {
+			label = "WLAN";
+			gpios = <&tlmm 5 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		reset {
+			label = "Reset";
+			gpios = <&tlmm 0 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_RESTART>;
+			linux,input-type = <1>;
```

Comment:
I think this can be dropped because Linux uses `<1>` as a default value.
https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/input/gpio-keys.yaml#L42

Reply:
Done